src/js/components/tasks/createForm.tsx

```diff
@@ -117,10 +117,18 @@ const CreateTaskModal = ({
         if (
           type === 'CREATE_OBJECT_SUCCEEDED' &&
           objectType === OBJECT_TYPES.TASK &&
-          object?.slug
+          object?.slug &&
+          epic
         ) {
           const url = routes.task_detail(project.slug, epic.slug, object.slug);
           history.push(url, { [RETRIEVE_CHANGES]: true });
+        } else if (
+          type === 'CREATE_OBJECT_SUCCEEDED' &&
+          objectType === OBJECT_TYPES.TASK &&
+          object?.slug
+        ) {
+          const url = routes.project_detail(project.slug);
+          history.push(url, { [RETRIEVE_CHANGES]: true });
```

Comment:
I don't think this block is necessary. This callback is only used when converting a Scratch Org into a new Task.